ASIM/dev/ASimTester/filteringTest/ASimFilteringTest.py

```diff
@@ -307,81 +306,168 @@ def get_values_for_dynamic_tests(self, rows):
-
+    def has_any_test(self, parameter_name, query_definition, no_filter_rows, column_name_in_table):
+        """
+        Test for dynamic parameters with a name that ends with "has_any". Filtering is made with substrings of values from no_filter_rows.   
+        
+        Parameters
+        ----------
+        parameter_name : Name of a parser's parameter
+        query_definition : A definition of the parser's query
+        no_filter_rows : The rows from a response for the parser query with no filter applied
+        column_name_in_table : The name of the column in the query response on which the parameter performs filtering
+        """
+        # Getting substrings that will be the values of the filtering parameters
+        selected_substrings = self.get_substrings_list(no_filter_rows, 2)
+        with self.subTest():
+            self.dynamic_tests_helper(parameter_name, query_definition, len(no_filter_rows), column_name_in_table, selected_substrings, "has_any")
+        
 
-    def add_prefix_to_list(self, rows, prefix_list, num_of_prefixes):
+    def get_prefix_list(self, rows, num_of_prefixes):
         '''
         The function return a list with at most "num_of_prefixes" prefixes of values from "rows" to "prefix_list".
         A prefix of a value will be the prefix until the first dot in the value (including the dot).
         '''
-        copy_prefix_list = prefix_list[:]
+        prefix_list = []
         # Looking for values with prefix that can be appended to the list
         for row in rows:
-            if len(copy_prefix_list) == num_of_prefixes:
+            if len(prefix_list) == num_of_prefixes:
                 break
     
             value = row[0]
-            pre = get_prefix(value, rows, copy_prefix_list)
+            pre = get_prefix(value, rows, prefix_list)
             # pre will equal value if: value dont contain a dot, pre is in the list, pre is contained in an item in the list.
             if pre != value:
-                copy_prefix_list.append(f"{pre}.")
+                prefix_list.append(f"{pre}.")
     
-        return copy_prefix_list
+        return prefix_list
+
+
+    def has_any_prefix_test(self, parameter_name, query_definition, no_filter_rows, column_name_in_table):
+        """
+        Test for dynamic parameters with a name that ends with "has_any_prefix". Filtering is made with prefixes of values from no_filter_rows.   
+        
+        Parameters
+        ----------
+        parameter_name : Name of a parser's parameter
+        query_definition : A definition of the parser's query
+        no_filter_rows : The rows from a response for the parser query with no filter applied
+        column_name_in_table : The name of the column in the query response on which the parameter performs filtering
+        """
+        # Getting prefixes that will be the values of the filtering parameters
+        selected_prefixes = self.get_prefix_list(no_filter_rows, 2)
```

Comment:
please add a comment that explains this '2'.
or extract to a variable with meaningful name